src/Sylius/Component/Core/Shipping/Checker/Rule/OrderTotalRuleChecker.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Shipping\Checker\Rule;
+
+use Sylius\Component\Core\Model\ShipmentInterface;
+use Sylius\Component\Shipping\Checker\Rule\RuleCheckerInterface;
+use Sylius\Component\Shipping\Model\ShippingSubjectInterface;
+
+abstract class OrderTotalRuleChecker implements RuleCheckerInterface
+{
+    public function isEligible(ShippingSubjectInterface $subject, array $configuration): bool
+    {
+        if (!$subject instanceof ShipmentInterface) {
+            return false;
+        }
+
+        $order = $subject->getOrder();
+        if (null === $order) {
+            return false;
+        }
+
+        $channel = $order->getChannel();
+        if (null === $channel) {
+            return false;
+        }
+
+        $amount = $configuration['total'][$channel->getCode()]['amount'] ?? null;
+        if (null === $amount) {
+            return false;
+        }
+
+        return $this->compare($order->getTotal(), $amount);
```

Comment:
🚨 

As far as I've investigated, this line is braking the scenario. The reason is when the cart is on the shipping selection step, it already has a default shipping method included which fee is a part of the order's total 🤯 So I suppose we should use `getItemsTotal()` here

Reply:
@Zales0123  / @loevgaard Can one of you please tell me how this issue is resolved? I think it can still be a problem. 
For example:
Set two shipping rules:
Method A. Order total less than or equal to 99.99 with the shipping costs of 5.00
Method B. Order total greater than or equal to 100 with the shipping costs of 0.00

Create a cart with an items total of 97.50.
Method A can't be used because the order total is now 102.50 (97.50 items total + 5.00 shipping). 
Method B can't be used either because, with the included shipping costs of this method, the order total of 100 (97.50 items total + 0 shipping) is not reached.

Reply:
I agree that it would probably make more sense that the shipping costs are not included in what we consider the order total for these rules.

Could you make an issue instead @MyUtd4Life?